### PR TITLE
Ruby 2.7 support

### PR DESCRIPTION
**Pull request by JonRowe**
These are some updates, generated from rspec-dev's rake tasks.

### CODE REVIEW

**Comment by JonRowe** [comment]
A couple of changes required to get a green 2.7.0 build, will probably cut a patch release with the changes.

File: lib/rspec/core/option_parser.rb [hunk 1]
```diff
@@ -22,8 +22,12 @@ def parse(source=nil)
...
         parser(options).parse!(args)
       rescue OptionParser::InvalidOption => e
-        failure = e.message
-        failure << " (defined in #{source})" if source
+        failure =
+          if source
+            "#{e.message} (defined in #{source})"
+          else
+            e.message
+          end
```
Comment (pirj) [inline lib/rspec/core/option_parser.rb]: Maybe:
```ruby
source_location = " (defined in #{source})" if source
abort "#{e.message}#{source_location}\n\n...
```

Comment (JonRowe) [reply lib/rspec/core/option_parser.rb]: Refactored a bit

Comment (JonRowe) [reply lib/rspec/core/option_parser.rb]: Can you take a look at rspec/rspec-support#392

**Comment by benoittgt** [comment]
Thanks!

**Pull request merged**